tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -410,12 +387,57 @@ void populateHLOToLHLOConversionPattern(MLIRContext* context,
   // clang-format on
 }
 
+/// Removes Lhlo.CopyOp that copies from an allocated buffer to the block
+/// argument. All uses of the buffer are replaced with the block argument.
+struct RedundantCopiesRemoval : mlir::FunctionPass<RedundantCopiesRemoval> {
+  void runOnFunction() override {
+    llvm::SmallVector<mlir::Operation*, 2> eraseList;
+    getFunction().walk([&](mlir::xla_lhlo::CopyOp copyOp) {
+      auto arguments = copyOp.getOperation()->getBlock()->getArguments();
```

Comment:
(transmitting @joker-eph): This assumes that the operation won't be in a block with implicit capture, I don't think we should have such assumption in passes.

Reply:
This is no longer required.